File: src/FizzBuzz.cpp
#include <iostream>

int main() {
    int fizz, buzz, min_val, max_val;
    std::cout << "Enter the Fizz value: ";
    std::cin >> fizz;
    std::cout << "Enter the Buzz value: ";
    std::cin >> buzz;
    std::cout << "Enter the minimum value: ";
    std::cin >> min_val;
    std::cout << "Enter the maximum value: ";
    std::cin >> max_val;
    
    for (int i = min_val; i <= max_val; i++) {
        if (i % fizz == 0 && i % buzz == 0) {
            std::cout << "FizzBuzz" << std::endl;
        } else if (i % fizz == 0) {
            std::cout << "Fizz" << std::endl;
        } else if (i % buzz == 0) {
            std::cout << "Buzz" << std::endl;
        } else {
            std::cout << i << std::endl;
        }
    }
    return 0;
}

File: src/FizzBuzz.js
function runFizzBuzz(fizz, buzz, minVal, maxVal) {
  const result = fizzBuzz(fizz, buzz, minVal, maxVal);
  return Promise.resolve(result);
}

function fizzBuzz(fizz, buzz, minVal, maxVal) {
  let output = '';
  for (let i = minVal; i <= maxVal; i++) {
    if (i % fizz === 0 && i % buzz === 0) {
      output += 'FizzBuzz ';
    } else if (i % fizz === 0) {
      output += 'Fizz ';
    } else if (i % buzz === 0) {
      output += 'Buzz ';
    } else {
      output += i + ' ';
    }
  }
  return output.trim();
}
async function generate() {
  const fizz = parseInt(document.getElementById("fizz").value);
  const buzz = parseInt(document.getElementById("buzz").value);
  const minVal = parseInt(document.getElementById("minVal").value);
  const maxVal = parseInt(document.getElementById("maxVal").value);
  
  const result = await runFizzBuzz(fizz, buzz, minVal, maxVal);
  console.log(result);
}

export default runFizzBuzz;
